docs/fitting.rst

```diff
@@ -15,6 +15,197 @@ initial guesses.  This model is then actually fit to the spectrum's ``flux``,
+      4.572864321608041  emission                91
+      4.824120603015076  emission                96
+      5.477386934673367  emission               109
+       8.99497487437186  emission               179
+
+   >>> absorption_lines = lines[lines['line_type'] == 'absorption'] #doctest:+SKIP
+         line_center    line_type  line_center_index
+              um
+      ----------------- ---------- -----------------
+      8.190954773869347 absorption               163
+
+An example using the `~specutils.fitting.find_lines_derivative`:
+
+.. code-block:: python
+
+   >>> # Define a noise region for adding the uncertainty
+   >>> noise_region = SpectralRegion(0*u.um, 3*u.um)
+
+   >>> # Derivative technique
+   >>> lines = find_lines_derivative(spectrum, flux_threshold=0.75)
+
+   >>> emission_lines = lines[lines['line_type'] == 'emission'] #doctest:+SKIP
+         line_center    line_type line_center_index
+              um                                   
+      ----------------- --------- -----------------
+      4.522613065326634  emission                90
+      5.477386934673367  emission               109
+
+   >>> absorption_lines = lines[lines['line_type'] == 'absorption'] #doctest:+SKIP
+         line_center    line_type  line_center_index
+              um                                    
+      ----------------- ---------- -----------------
+      8.190954773869347 absorption               163
+
+
+While it might be surprising that these tables do not contain more information
+about the lines, this is because the "toolbox" philosophy of ``specutils`` aims to
+keep such functionality in separate distinct functions.  See :doc:`analysis` for
+functions that can be used to fill out common line measurements more
+completely.
+
+Parameter Estimation
+--------------------
+
+Given a spectrum with a set of lines, the `~specutils.fitting.estimate_line_parameters`
+can be called to estimate the `~astropy.modeling.Model` parameters given a spectrum.
+
+For the `~astropy.modeling.functional_models.Gaussian1D`, 
+`~astropy.modeling.functional_models.Voigt1D`, and
+`~astropy.modeling.functional_models.Lorentz1D` models, there are predefined estimators for each 
+of the parameters. For all other models one must define the estimators (see example below).
+Note that in many (most?) cases where another model is needed, it may be better to create
+your own template models tailored to your specific spectra and skip this function entirely.
+
+
+For example, based on the spectrum defined above:
+
+.. code-block:: python
+
+   >>> from specutils import SpectralRegion
+   >>> from specutils.fitting import estimate_line_parameters
+   >>> from specutils.manipulation import extract_region
+
+   >>> sub_region = SpectralRegion(4*u.um, 5*u.um)
+   >>> sub_spectrum = extract_region(spectrum, sub_region)
+
+   >>> # Estimate parameter Gaussian1D
+   >>> g_init = estimate_line_parameters(sub_spectrum, models.Gaussian1D()) #doctest:+SKIP
```

Comment:
Why skip the doctest here?  Definitely we want to check this so the user sees the right thing!
```suggestion
   >>> g_init = estimate_line_parameters(sub_spectrum, models.Gaussian1D())
   >>> g_init
```